src/core/cgroup-bpf-ctx.c

```diff
@@ -1,22 +1,83 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
 
+#include "bpf-program.h"
 #include "cgroup-bpf-ctx.h"
 #include "cgroup.h"
 #include "memory-util.h"
 #include "strv.h"
 #include "string-util.h"
 #include "path-util.h"
+#include "strv.h"
+
+int bpffs_program_from_string(const char *str, enum bpf_attach_type *ret_attach_type, char **ret_bpffs_path) {
+        _cleanup_strv_free_ char **parts = NULL;
+        _cleanup_free_ char *dup = NULL;
+        int attach_type;
+
+        assert(str);
+        assert(ret_bpffs_path);
+        assert(ret_attach_type);
+
+        parts = strv_split(str, ":");
+        if (!parts)
+                return log_oom();
+
+        if (strv_length(parts) < 2)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Unexpected BPF program string format %s", str);
+
+        dup = strdup(parts[1]);
+        if (!dup)
+                return log_oom();
```

Comment:
there should be some validation here. i.e. probably a path_is_normalized() check, and at least a path_is_valid() check

Reply:
addressed by adding `path_is_normalized`